Add bower and linters support

comzeradd opened the pull request:
* Update templates to use new libraries paths
* Update docker to use bower

comzeradd commented:
I amended the commit to check in the static libraries. 

Both options have merits and you'll find projects on either paths. The primary benefit of having everything in our repo is that we don't rely on an external source during the deployment (eg. what bower registry or github are down during a deployment cycle). The main obvious disadvantage is that we increase the size of our repos since we pull the entire frontend packages, not just the minified production files we need. (an [interesting read](https://addyosmani.com/blog/checking-in-front-end-dependencies/) on that topic)

In general I have a slight preference of not checking in frontend libraries, but since we move to a more CD workflow it makes more sense for our projects to minimize risk of the deployment process.

johngian commented:
@comzeradd Regarding checking vs ignoring files I think that we should find a middleground

We shouldn't rely on bower to fetch dependencies on deploy time but we should also avoid maintaining so many files. Have you researched any best practices? Maybe using gulp and have a task to fetch specific files after bower install?

johngian commented on Dockerfile:
```diff
@@ -7,8 +7,10 @@ CMD ["./bin/run-prod.sh"]
+    apt-get install -y --no-install-recommends build-essential libmysqlclient-dev mysql-client npm git && \
     rm -rf /var/lib/apt/lists/*
+RUN npm install -g bower
+RUN ln -s /usr/bin/nodejs /usr/bin/node
```
Instead of manually linking ``nodejs`` and ``node`` legacy binaries you can always install the latest version:
https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions

johngian commented:
Regarding workarounds. I found those:

https://github.com/ck86/main-bower-files
https://www.npmjs.com/package/gulp-main-bower-files

The idea is that:
* We ignore bower components folder
* We install bower dependencies via gulp
* Add a task to fetch the *main* files of each css/js package
* Store the files to a version controlled folder

comzeradd commented:
@johngian Nice tip! I added gulp and combined it with uglify and clean-css to minify. Now we only check in the files we actually need.

To test it locally you first need to have bower and gulp-cli (probably through npm).
```
sudo npm install -g bower gulp-cli
```

And then just install all dependencies and run gulp.
```
npm install
bower install
gulp
```

comzeradd commented:
I added one more commit to add linters for both js and css code, since we can use gulp for these tests locally too. I also did some research on figuring out the proper code style rules, generally as best practices but also what other webdev teams inside Mozilla use.

johngian commented:
Please add some docs on how to set this up.

johngian commented:
I suggest we also add the build dependencies to docker

* It's not that many
* We (at least me 😄) are still using docker for local development

johngian commented on gulpfile.js:
```diff
@@ -0,0 +1,52 @@
+    'moderator/moderate/static/css/*.css'
+];
+
+gulp.task('js:lint', () => {
```
It was an old version, using latest works fine.

johngian commented:
```
$ ./manage.py collectstatic
...
...
whitenoise.storage.MissingFileError: The file 'lib/mozilla-tabzilla/media/img/tabzilla-static-high-res.png' could not be found with <whitenoise.storage.CompressedManifestStaticFilesStorage object at 0x7f0af23b4350>.

The CSS file 'lib/mozilla-tabzilla/css/tabzilla.css' references a file which could not be found:
  lib/mozilla-tabzilla/media/img/tabzilla-static-high-res.png

Please check the URL references in this CSS file, particularly any
relative paths which might be pointing to the wrong location.
```

comzeradd commented:
PR Update
* I fixed the missing png file
* Added some basic documenation on README
* Added necessary tools on Dockerfile. I didn't choose to go with latest nodejs, since I get dependency conflicts with npm (or debian is messing with me :D)

johngian commented:
OK i will take a look about nodejs and test locally. Other than that looks good.

johngian commented:
I think that the commands from the install page do the trick:

https://github.com/mozilla/mozmoderator/commit/609561b2232685c12639ba18878264755daa38cd

comzeradd commented:
@johngian I cherry-picked your commit.
